File: client/src/pages/Grade.js
import { Card } from "semantic-ui-react";

const Grade = ({ score, id, user_name, skill_name, user_id, skill_id }) => {
  return (
    <Card>
      <Card.Content header={`${user_name}`} />
      <Card.Content description={`score: ${score} on ${skill_name}`} />
      <Card.Content extra>
        <p>
          user id: {user_id} skill id: {skill_id}
        </p>
      </Card.Content>
    </Card>
  );
};

export default Grade;

File: app/models/grade.rb
class Grade < ApplicationRecord
  belongs_to :user
  belongs_to :skill

  def self.all_and_then_some
    # SQL CALL HERE (NOT TODAY)

    # rails way

    # good old fashion ruby (bad way to do but works)
    grades = Grade.all
    grades.map do |grade|
      { id: grade.id, score: grade.score, user_name: grade.user.name, skill_name: grade.skill.name, user_id: grade.user.id, skill_id: grade.skill.id }
    end
  end
end
